examples/3dsphere/deformation_flow.jl

```diff
@@ -0,0 +1,214 @@
+const λ_c2 = 210.0 # initial longitude of second tracer
+const ϕ_c = 0.0 # initial latitude of tracers
+const z_c = 5.0e3 # initial altitude of tracers
+const center1 = Geometry.LatLongPoint(ϕ_c, λ_c1)
+const center2 = Geometry.LatLongPoint(ϕ_c, λ_c2)
+const R_t = R / 2 # horizontal half-width of tracers
+const Z_t = 1000.0 # vertical half-width of tracers
+
+# set up 3D domain - doubly periodic box
+hv_center_space, hv_face_space = sphere_3D(R)
+
+# Φ(z) = grav * z
+
+# # initial conditions
+coords = Fields.coordinate_field(hv_center_space)
+face_coords = Fields.coordinate_field(hv_face_space)
+
+r1(λ, ϕ) = R * acos( sind(ϕ_c)*sind(ϕ) + cosd(ϕ_c)*cosd(ϕ)*cosd(λ-λ_c1) )
+r2(λ, ϕ) = R * acos( sind(ϕ_c)*sind(ϕ) + cosd(ϕ_c)*cosd(ϕ)*cosd(λ-λ_c2) )
+
+# d1(λ, ϕ, z) = minimum(1, (r1(λ, ϕ)/R_t)^2+((z-z_c)/Z_t)^2)
+# d2(λ, ϕ, z) = minimum(1, (r2(λ, ϕ)/R_t)^2+((z-z_c)/Z_t)^2)
+
+p(z) = p_0 * exp(-z/H)
+ρ(z) = p(z)/R_d/T_0
+
+y0 = map(coords) do coord
+    # rd1 = Geometry.great_circle_distance(coord, center1, hv_center_space.global_geom)
+    # rd2 = Geometry.great_circle_distance(coord, center2, hv_center_space.global_geom)
+    z = coord.z
+    zd = z - z_c
+    λ = mod(coord.long, 360)
+    ϕ = coord.lat
+    rd1 = r1(λ, ϕ)
+    rd2 = r2(λ, ϕ)
+
+    d1 = min(1, (rd1/R_t)^2 + (zd/Z_t)^2)
+    d2 = min(1, (rd2/R_t)^2 + (zd/Z_t)^2)
+    
+    q1 = 1/2.0*(1 + cos(pi*d1)) + 1/2.0*(1 + cos(pi*d2)) 
+    q2 = 0.9 - 0.8 * q1^2
+    q3 = 0.0
+    if d1 < 0.5 || d2 < 0.5
+        q3 = 1.0
+    else
+        q3 = 0.1
+    end
+    
+    if (z > z_c) && (ϕ > ϕ_c - rad2deg(1/8)) && (ϕ < ϕ_c + rad2deg(1/8))
+       q3 = 0.1
+    end
+    q4 = 1 - 3/10*(q1+q2+q3)
+
+    ρq1 = ρ(z)*q1
+    ρq2 = ρ(z)*q2
+    ρq3 = ρ(z)*q3
+    ρq4 = ρ(z)*q4
+    
+    return (ρq1=ρq1, ρq2=ρq2, ρq3=ρq3, ρq4=ρq4)
+    
+end
+
+
+# # Vertical momentum defined on cell faces
+# ρw = map(face_coords) do coord
+#     Geometry.WVector(0.0)
+# end;
+
+# Y = Fields.FieldVector(Yc = Yc, ρw = ρw)
+
+function rhs!(dydt, y, coords, t)
+    
+    ϕ = coords.lat
+    λ = mod.(coords.long, 360)
+    z = coords.z
+    λp = λ .- 360*t/τ
+    k = 10*R/τ 
+
+    sp = 1 + exp( (p_top-p_0)/b/p_top ) .- exp.( (p.(z) .- p_0)./b./p_top ) .- exp.( (p_top .- p.(z))./b./p_top )
+
+    uu = k .* sind.(λp).^2  .* sind.(2 .* ϕ) .* cos(pi*t/τ) .+ 2*pi*R/τ .* cosd.(ϕ)
+    uv = k .* sind.(2 .* λp).^2  .* cosd.(ϕ) .* cos(pi*t/τ)
+    uw = -ω_0 .* sind.(λp) .* cosd.(ϕ) .* cos(2*pi*t/τ) .* sp ./ ρ.(z) ./ grav
+
+    uₕ = Geometry.UVVector.(uu, uv)  
+    w = Geometry.WVector.(uw)
+
+    If = Operators.InterpolateC2F(
+        bottom = Operators.Extrapolate(),
+        top = Operators.Extrapolate(),
+    )
+
+    ∂ = Operators.DivergenceF2C(
+        bottom = Operators.SetValue(Geometry.WVector(0.0)),
+        top = Operators.SetValue(Geometry.WVector(0.0)),
+    )
+
+    hdiv = Operators.Divergence()
+
+    @. dydt.ρq1   = -∂(If(y.ρq1 .* w))
+    @. dydt.ρq1  -= hdiv(y.ρq1 .* uₕ)
```

Comment:
@simonbyrne Is this correct for scalar advection? uh and w are physical velocities and are on cell centers.